### DISCUSSION TITLE
Maya-Arnold material export

### DISCUSSION THREAD

**Discussion by jcoelho88** [Autodesk/maya-usd#1657] [Q&A] [answered]
Hello,

I have a question about how the current Maya-usd export handles exporting Looks that contain both Maya and Arnold nodes.

Setup: Maya 2022.1, Maya-usd 0.10.0, Arnold 4.2.4

I did a test with a sphere, its got one aiStandardSurface with a Maya file node connected.

In the test I rendered the object as created in Maya (left), then an imported object (USD import) of the USD exported object (with the material option set to Arnold)(Center) and finally the same USD exported object loaded into USD Maya stage (right).

![image](https://user-images.githubusercontent.com/88895104/130995376-1e6541a8-63ac-4e2f-8269-28be6ac32929.png)

My question is why is there a difference in the results for the last object?

I apologize if there is already an answer out there that I missed. From what I can gather, is it the intention that if you use a Maya node like the file node that you are expected to be doing a round trip back into Maya by importing it thus converting from USD back into Maya data?

If I want to keep my data as USD should I not be using any Maya specific nodes and stick to just Arnold ones?

Any insight into this would be appreciated.

Cheers,

Jason Coelho

**Comment by santosd** [comment]
Hello @jcoelho88,

This is a workflow that is still in progress and not completed. Currently there are two different ways to export Arnold Shaders to USD data. One is through the File > Export menu using Maya USD Export and selecting Arnold Surface Shader for the material. This allows import and export of the of the materials but doesn't translate to the stage.

There is also an export through Arnold > Scene Export > Export Selection to USD... and this one does translate to the stage in an Arnold render.

We are working on merging these two into one that covers most workflows.

I hope that information is helpful in understanding the issue.

Cheers,

David Santos

**Reply by jcoelho88** [reply]
Hi @santosd,

Thanks for the explanation, I do understand the two methods and I'm glad the hear that there is still development happening.

Cheers,

Jason Coelho

**Comment by sebastienblor** [comment, marked as answer]
Hi Jason, 
indeed this is an expected limitation. If you export maya shaders to usd with the "Arnold Materials" option enabled, it will export these maya shaders in a specific format that is only recognized by the import. So if you load the same file in a usd stage, these shaders won't be recognized. We're currently improving this, so that both options work. 

Note that when this will be implemented, exoprting maya shaders with the "Arnold Materials" option will export these shaders as a combination of arnold shading nodes. So when you'll import them back, they will still render as before, but you will see different shading nodes in your scene. 
Cheers

**Comment by jcoelho88** [comment]
Hi @sebastienblor,

Thanks for the explanation, I understand the Arnold Materials option would essentially turn Maya nodes to Arnold equivalent ones on export/import and I think that should be fine. The most important aspect that I was hoping for is that these materials would be be recognized in a USD stage with Arnold in any DCC that implements the Renderer for USD.

Ill close this discussion with this answer.

Cheers,

Jason Coelho
